File: frontend/src/app/models/Video.ts
import { User } from './User';

export interface Video {
  id: number;
  title: string;
  description: string;
  videoFile: string;
  thumbnail: string;
  published: boolean;
  duration: number; // seconds
  views: number | null;
  createdAt: Date;
  updatedAt: Date;
  owner: User;
}

File: backend/src/main/java/com/mthree/backend/models/Video.java
package com.mthree.backend.models;

import java.sql.Date;
import java.sql.Timestamp;

import org.hibernate.annotations.CreationTimestamp;
import org.hibernate.annotations.UpdateTimestamp;

import jakarta.persistence.*;
import lombok.Data;
import lombok.NoArgsConstructor;
import lombok.RequiredArgsConstructor;

@Entity
@Table(name = "videos")
@Data
@NoArgsConstructor
public class Video {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long id;

    private String videoFile;

    private String thumbnail;

    private String title;

    private String description;

    private long duration;

    private int views;

    private boolean isPublished;

    @ManyToOne @JoinColumn(name = "owner")
    private User owner;

    @CreationTimestamp
    private Timestamp createdAt;

    @UpdateTimestamp
    private Timestamp updatedAt;
}

